### PR TITLE
pkp/pkp-lib#700 Clean up plugin API

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: classes/controllers/grid/plugins/PKPPluginGridRow.inc.php [hunk 1]
```diff
@@ -155,9 +100,7 @@ function initialize($request, $template = null) {
 	 * @param $plugin Plugin
 	 * @return boolean
 	 */
-	function canEdit(&$plugin) {
-		assert(false); // Must be overridden by subclasses
-	}
+	protected abstract function _canEdit($plugin);
```
Comment (beghelli) [inline classes/controllers/grid/plugins/PKPPluginGridRow.inc.php]: protected with underscore?

File: controllers/grid/files/SubmissionFilesGridDataProvider.inc.php [hunk 2]
```diff
@@ -132,7 +132,7 @@ function getAddFileAction($request) {
 	 * @return array
 	 */
 	protected function applyFilter($revisions, $filter) {
-		if (isset($filter['search'])) switch ($filter['column']) {
+		if (!empty($filter['search'])) switch ($filter['column']) {
```
Comment (beghelli) [inline controllers/grid/files/SubmissionFilesGridDataProvider.inc.php]: From another issue right?

File: plugins/generic/usageStats/PKPUsageStatsPlugin.inc.php [hunk 3]
```diff
@@ -103,39 +102,27 @@ function getTemplatePath($inCore = false) {
...
-	* @see PKPPlugin::manage()
-	*/
-	function manage($verb, $args, &$message, &$messageParams, &$pluginModalContent = null) {
-		$returner = parent::manage($verb, $args, $message, $messageParams);
-		if (!$returner) return false;
-
-		$request = $this->getRequest();
+	 * @see PKPPlugin::manage()
+	 */
+	function manage($request, $verb, $args) {
```
Comment (beghelli) [inline plugins/generic/usageStats/PKPUsageStatsPlugin.inc.php]: wrong arguments, see Plugin::manage()

File: plugins/generic/usageStats/PKPUsageStatsPlugin.inc.php [hunk 4]
```diff
@@ -103,39 +102,27 @@ function getTemplatePath($inCore = false) {
...
+					$notificationManager->createTrivialNotification($user->getId(), NOTIFICATION_TYPE_SUCCESS, array('contents' => __('plugins.generic.usageStats.settings.saved')));
+					return true;
 				}
-				break;
-			default:
-				return $returner;
+				return new JSONMessage(true, $settingsForm->fetch($request));
 		}
-		return true;
+		return parent::manage($request, $verb, $args);
```
Comment (beghelli) [inline plugins/generic/usageStats/PKPUsageStatsPlugin.inc.php]: same as above

File: plugins/generic/usageStats/PKPUsageStatsPlugin.inc.php [hunk 5]
```diff
@@ -143,32 +130,25 @@ function manage($verb, $args, &$message, &$messageParams, &$pluginModalContent =
...
-				$router->url($request, null, null, 'plugin', null, array('verb' => 'settings', 'plugin' => $this->getName(), 'category' => 'generic')),
-				$this->getDisplayName()
-			);
-			return new LinkAction($verbName, $actionRequest, $verbLocalized, null);
-		}
+		import('lib.pkp.classes.linkAction.request.AjaxModal');
+		return array_merge(
+			array(
+				new LinkAction(
+					$verbName,
```
Comment (beghelli) [inline plugins/generic/usageStats/PKPUsageStatsPlugin.inc.php]: $verbName is not defined anymore I think

File: controllers/grid/files/SubmissionFilesGridDataProvider.inc.php (hunk 2 above)
Comment (asmecher) [inline controllers/grid/files/SubmissionFilesGridDataProvider.inc.php]: Oops! Yes.

**Pull request merged**